### ISSUE TITLE
ReduceLROnPlateau forced casting on integer learning rates

### ISSUE THREAD

**Issue by alipwong** [tensorflow/tensorflow#45261]
**Issue in `ReduceLROnPlateau`**
- `ReduceLROnPlateau` relies of the function: `Keras.set_value`
- When using `ReduceLROnPlateau`, this ensures the reduced LR is of the same type as input LR
- This is an issue if the learning rate is specified as an integer. I.e. different behaviour for LR = 1 and LR = 1.0

**Describe the current behavior**
Example 1: Initial learning rate 1, factor 0.5. First reduction should drop to 0.5.
Example 2:Initial learning rate 5, factor 0.5. First reduction should drop to 2.5.

**Describe the expected behavior**
Example 1: First reduction drops to 0.
Example 2: First reduction drops to 2.

**Standalone code to reproduce the issue**
```
import numpy as np
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras.optimizers import Adam
from tensorflow.keras.callbacks import ReduceLROnPlateau

n = 100

model = keras.Sequential()
model.add(keras.layers.Dense(n, activation='linear')

model.compile(optimizer = keras.optimizers.Adam(lr=1), loss =  ['mse'])

x = np.random.rand(2, n)
y = np.random.rand(2, n)

reduce_lr = ReduceLROnPlateau(monitor='loss', min_lr=1e-6, patience = 1, factor = 0.1, verbose = 1)

history = model.fit(x, y, epochs = 10, callbacks = [reduce_lr,], verbose = 0)

history.history['lr']
```
**Outputs**
Epoch 00002: ReduceLROnPlateau reducing learning rate to 0.1.

[1, 1, 0, 0, 0, 0, 0, 0, 0, 0]

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.4, nightly version(`2.5.0-dev20201129`) and was able to reproduce the issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/6eeb6765af29ffb7cd0e3519be0473d4/untitled550.ipynb).Thanks!

**Comment by jvishnuvardhan** [comment]
@alipwong Thanks for the clear report. I can reproduce the issue. However, the `learning_rate` is supposed to be a floating value as mentioned in the [here](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adam). Once we change learning rate from int to float, everything works as expected. [Here](https://colab.research.google.com/gist/jvishnuvardhan/bf1a3247a19ab898ded29bb8975a7efe/untitled550.ipynb) is the gist for our reference.

I am not sure whether it is intentional or not. I see that `k.set_value` used in other callbacks also. we will look into it. Thanks!

**Comment by saikumarchalla** [comment]
@alipwong  Could you please respond t @jvishnuvardhan comment.Thanks!
